[WebXR] Port PlatformXR::FrameData to new serialization format

djg opened the pull request:
#### 9a4085b2914ed55f4dd95684333d61c8511faa14
<pre>
[WebXR] Port PlatformXR::FrameData to new serialization format
<a href="https://bugs.webkit.org/show_bug.cgi?id=266202">https://bugs.webkit.org/show_bug.cgi?id=266202</a>
<a href="https://rdar.apple.com/119474838">rdar://119474838</a>

Reviewed by Brady Eidson.

Port PlatformXR::FrameData and it&apos;s related member structs to use automatically
generated IPC serialization and remove the explicit encode/decode functions.

Due to constraints imposed by generate-serializers.py,
PlatformXR::Device::FrameData has been moved to PlatformXR::FrameData.

* Source/WebCore/Modules/webxr/WebXRFrame.cpp:
(WebCore::WebXRFrame::getViewerPose):
(WebCore::WebXRFrame::matrixFromPose):
* Source/WebCore/Modules/webxr/WebXRFrame.h:
* Source/WebCore/Modules/webxr/WebXRGamepad.cpp:
(WebCore::WebXRGamepad::WebXRGamepad):
* Source/WebCore/Modules/webxr/WebXRGamepad.h:
* Source/WebCore/Modules/webxr/WebXRHand.cpp:
(WebCore::WebXRHand::updateFromInputSource):
* Source/WebCore/Modules/webxr/WebXRHand.h:
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::create):
(WebCore::WebXRInputSource::WebXRInputSource):
(WebCore::WebXRInputSource::update):
* Source/WebCore/Modules/webxr/WebXRInputSource.h:
* Source/WebCore/Modules/webxr/WebXRInputSourceArray.h:
* Source/WebCore/Modules/webxr/WebXRInputSpace.cpp:
(WebCore::WebXRInputSpace::create):
(WebCore::WebXRInputSpace::WebXRInputSpace):
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
(WebCore::WebXRInputSpace::setPose):
* Source/WebCore/Modules/webxr/WebXRJointSpace.cpp:
(WebCore::WebXRJointSpace::create):
(WebCore::WebXRJointSpace::WebXRJointSpace):
(WebCore::WebXRJointSpace::updateFromJoint):
* Source/WebCore/Modules/webxr/WebXRJointSpace.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::WebXROpaqueFramebuffer::startFrame):
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.h:
* Source/WebCore/Modules/webxr/WebXRSession.cpp:
(WebCore::WebXRSession::sessionDidInitializeInputSources):
(WebCore::WebXRSession::onFrame):
* Source/WebCore/Modules/webxr/WebXRSession.h:
* Source/WebCore/Modules/webxr/WebXRSystem.cpp:
(WebCore::WebXRSystem::DummyInlineDevice::requestFrame):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.cpp:
(WebCore::WebXRWebGLLayer::startFrame):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.h:
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::FrameData::copy const):
(PlatformXR::Device::FrameData::FloatQuaternion::encode const): Deleted.
(PlatformXR::Device::FrameData::FloatQuaternion::decode): Deleted.
(PlatformXR::Device::FrameData::Pose::encode const): Deleted.
(PlatformXR::Device::FrameData::Pose::decode): Deleted.
(PlatformXR::Device::FrameData::Fov::encode const): Deleted.
(PlatformXR::Device::FrameData::Fov::decode): Deleted.
(PlatformXR::Device::FrameData::View::encode const): Deleted.
(PlatformXR::Device::FrameData::View::decode): Deleted.
(PlatformXR::Device::FrameData::StageParameters::encode const): Deleted.
(PlatformXR::Device::FrameData::StageParameters::decode): Deleted.
(PlatformXR::Device::FrameData::LayerData::encode const): Deleted.
(PlatformXR::Device::FrameData::LayerData::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourceButton::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourceButton::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourcePose::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourcePose::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourceHandJoint::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourceHandJoint::decode): Deleted.
(PlatformXR::Device::FrameData::InputSource::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSource::decode): Deleted.
(PlatformXR::Device::FrameData::encode const): Deleted.
(PlatformXR::Device::FrameData::decode): Deleted.
(PlatformXR::Device::FrameData::copy const): Deleted.
* Source/WebCore/platform/xr/cocoa/PlatformXRPose.cpp:
(PlatformXRPose::orientation const):
(PlatformXRPose::pose const):
* Source/WebCore/platform/xr/cocoa/PlatformXRPose.h:
* Source/WebCore/platform/xr/openxr/OpenXRInput.cpp:
(PlatformXR::OpenXRInput::collectInputSources const):
* Source/WebCore/platform/xr/openxr/OpenXRInput.h:
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.cpp:
(PlatformXR::OpenXRInputSource::getInputSource const):
(PlatformXR::OpenXRInputSource::getPose const):
(PlatformXR::OpenXRInputSource::getButton const):
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.h:
* Source/WebCore/platform/xr/openxr/OpenXRLayer.cpp:
(PlatformXR::OpenXRLayerProjection::startFrame):
* Source/WebCore/platform/xr/openxr/OpenXRLayer.h:
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h:
(PlatformXR::XrPosefToPose):
(PlatformXR::xrViewToPose):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp:
(PlatformXR::OpenXRDevice::requestFrame):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h:
* Source/WebCore/testing/WebFakeXRDevice.cpp:
(WebCore::FakeXRView::setFieldOfView):
(WebCore::SimulatedXRDevice::setViews):
(WebCore::SimulatedXRDevice::setViewerOrigin):
(WebCore::SimulatedXRDevice::frameTimerFired):
(WebCore::WebFakeXRDevice::setViews):
(WebCore::WebFakeXRDevice::parseRigidTransform):
* Source/WebCore/testing/WebFakeXRDevice.h:
* Source/WebCore/testing/WebFakeXRInputController.cpp:
* Source/WebCore/testing/WebFakeXRInputController.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/XR/PlatformXR.serialization.in: Added.
* Source/WebKit/Shared/XR/XRSystem.serialization.in:
* Source/WebKit/UIProcess/XR/PlatformXRSystem.cpp:
(WebKit::PlatformXRSystem::requestFrame):
* Source/WebKit/UIProcess/XR/PlatformXRSystem.h:
* Source/WebKit/UIProcess/XR/PlatformXRSystem.messages.in:
* Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm:
(WebKit::ARKitCoordinator::renderLoop):
* Source/WebKit/UIProcess/XR/ios/WKARPresentationSession.h:
* Source/WebKit/UIProcess/XR/ios/WKARPresentationSession.mm:
(-[_WKARPresentationSession collectInputSources]):
(-[_WKTransientGestureRecognizer _platformXRInputSourceFromTransientAction:actionIdentifier:]):
(-[_WKTransientGestureRecognizer collectInputSources]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: <a href="https://commits.webkit.org/271912@main">https://commits.webkit.org/271912@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/170f0e167b41a677e55770bdb5a64d24b87aa158

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/30040 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/8702 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/31350 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/32538 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/27153 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/30700 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/10894 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/5952 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/27185 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/30334 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/7293 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/25613 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/6225 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/30192 "Passed tests") | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/6382 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/26705 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/33873 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/27388 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/27138 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/32563 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/6327 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/4496 "Found 2 new test failures: fullscreen/full-screen-layer-dump.html, imported/w3c/web-platform-tests/css/css-masking/clip-path-svg-content/clip-path-content-use-007.svg (failure)") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/30361 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/8076 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/7113 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/7079 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/6854 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

beidson commented on Source/WebKit/WebKit.xcodeproj/project.pbxproj:
```diff
@@ -18853,7 +18855,6 @@
 			isa = PBXTargetDependency;
 			platformFilters = (
 				ios,
-				xros,
```
This and the other 3 xros subtractions seem errant.

djg replied:
Yeah, not sure what Xcode was doing...

beidson reviewed (approved):
Need to make the same structure changes over in openxr land to keep the other platforms happy.

Other than that LGTM

djg committed:
[WebXR] Port PlatformXR::FrameData to new serialization format
https://bugs.webkit.org/show_bug.cgi?id=266202
rdar://119474838

Reviewed by Brady Eidson.

Port PlatformXR::FrameData and it's related member structs to use automatically
generated IPC serialization and remove the explicit encode/decode functions.

Due to constraints imposed by generate-serializers.py,
PlatformXR::Device::FrameData has been moved to PlatformXR::FrameData.

* Source/WebCore/Modules/webxr/WebXRFrame.cpp:
(WebCore::WebXRFrame::getViewerPose):
(WebCore::WebXRFrame::matrixFromPose):
* Source/WebCore/Modules/webxr/WebXRFrame.h:
* Source/WebCore/Modules/webxr/WebXRGamepad.cpp:
(WebCore::WebXRGamepad::WebXRGamepad):
* Source/WebCore/Modules/webxr/WebXRGamepad.h:
* Source/WebCore/Modules/webxr/WebXRHand.cpp:
(WebCore::WebXRHand::updateFromInputSource):
* Source/WebCore/Modules/webxr/WebXRHand.h:
* Source/WebCore/Modules/webxr/WebXRInputSource.cpp:
(WebCore::WebXRInputSource::create):
(WebCore::WebXRInputSource::WebXRInputSource):
(WebCore::WebXRInputSource::update):
* Source/WebCore/Modules/webxr/WebXRInputSource.h:
* Source/WebCore/Modules/webxr/WebXRInputSourceArray.h:
* Source/WebCore/Modules/webxr/WebXRInputSpace.cpp:
(WebCore::WebXRInputSpace::create):
(WebCore::WebXRInputSpace::WebXRInputSpace):
* Source/WebCore/Modules/webxr/WebXRInputSpace.h:
(WebCore::WebXRInputSpace::setPose):
* Source/WebCore/Modules/webxr/WebXRJointSpace.cpp:
(WebCore::WebXRJointSpace::create):
(WebCore::WebXRJointSpace::WebXRJointSpace):
(WebCore::WebXRJointSpace::updateFromJoint):
* Source/WebCore/Modules/webxr/WebXRJointSpace.h:
* Source/WebCore/Modules/webxr/WebXRLayer.h:
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.cpp:
(WebCore::WebXROpaqueFramebuffer::startFrame):
* Source/WebCore/Modules/webxr/WebXROpaqueFramebuffer.h:
* Source/WebCore/Modules/webxr/WebXRSession.cpp:
(WebCore::WebXRSession::sessionDidInitializeInputSources):
(WebCore::WebXRSession::onFrame):
* Source/WebCore/Modules/webxr/WebXRSession.h:
* Source/WebCore/Modules/webxr/WebXRSystem.cpp:
(WebCore::WebXRSystem::DummyInlineDevice::requestFrame):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.cpp:
(WebCore::WebXRWebGLLayer::startFrame):
* Source/WebCore/Modules/webxr/WebXRWebGLLayer.h:
* Source/WebCore/platform/xr/PlatformXR.h:
(PlatformXR::FrameData::copy const):
(PlatformXR::Device::FrameData::FloatQuaternion::encode const): Deleted.
(PlatformXR::Device::FrameData::FloatQuaternion::decode): Deleted.
(PlatformXR::Device::FrameData::Pose::encode const): Deleted.
(PlatformXR::Device::FrameData::Pose::decode): Deleted.
(PlatformXR::Device::FrameData::Fov::encode const): Deleted.
(PlatformXR::Device::FrameData::Fov::decode): Deleted.
(PlatformXR::Device::FrameData::View::encode const): Deleted.
(PlatformXR::Device::FrameData::View::decode): Deleted.
(PlatformXR::Device::FrameData::StageParameters::encode const): Deleted.
(PlatformXR::Device::FrameData::StageParameters::decode): Deleted.
(PlatformXR::Device::FrameData::LayerData::encode const): Deleted.
(PlatformXR::Device::FrameData::LayerData::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourceButton::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourceButton::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourcePose::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourcePose::decode): Deleted.
(PlatformXR::Device::FrameData::InputSourceHandJoint::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSourceHandJoint::decode): Deleted.
(PlatformXR::Device::FrameData::InputSource::encode const): Deleted.
(PlatformXR::Device::FrameData::InputSource::decode): Deleted.
(PlatformXR::Device::FrameData::encode const): Deleted.
(PlatformXR::Device::FrameData::decode): Deleted.
(PlatformXR::Device::FrameData::copy const): Deleted.
* Source/WebCore/platform/xr/cocoa/PlatformXRPose.cpp:
(PlatformXRPose::orientation const):
(PlatformXRPose::pose const):
* Source/WebCore/platform/xr/cocoa/PlatformXRPose.h:
* Source/WebCore/platform/xr/openxr/OpenXRInput.cpp:
(PlatformXR::OpenXRInput::collectInputSources const):
* Source/WebCore/platform/xr/openxr/OpenXRInput.h:
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.cpp:
(PlatformXR::OpenXRInputSource::getInputSource const):
(PlatformXR::OpenXRInputSource::getPose const):
(PlatformXR::OpenXRInputSource::getButton const):
* Source/WebCore/platform/xr/openxr/OpenXRInputSource.h:
* Source/WebCore/platform/xr/openxr/OpenXRLayer.cpp:
(PlatformXR::OpenXRLayerProjection::startFrame):
* Source/WebCore/platform/xr/openxr/OpenXRLayer.h:
* Source/WebCore/platform/xr/openxr/OpenXRUtils.h:
(PlatformXR::XrPosefToPose):
(PlatformXR::xrViewToPose):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.cpp:
(PlatformXR::OpenXRDevice::requestFrame):
* Source/WebCore/platform/xr/openxr/PlatformXROpenXR.h:
* Source/WebCore/testing/WebFakeXRDevice.cpp:
(WebCore::FakeXRView::setFieldOfView):
(WebCore::SimulatedXRDevice::setViews):
(WebCore::SimulatedXRDevice::setViewerOrigin):
(WebCore::SimulatedXRDevice::frameTimerFired):
(WebCore::WebFakeXRDevice::setViews):
(WebCore::WebFakeXRDevice::parseRigidTransform):
* Source/WebCore/testing/WebFakeXRDevice.h:
* Source/WebCore/testing/WebFakeXRInputController.cpp:
* Source/WebCore/testing/WebFakeXRInputController.h:
* Source/WebKit/CMakeLists.txt:
* Source/WebKit/DerivedSources-input.xcfilelist:
* Source/WebKit/DerivedSources.make:
* Source/WebKit/Scripts/webkit/messages.py:
(headers_for_type):
* Source/WebKit/Shared/XR/PlatformXR.serialization.in: Added.
* Source/WebKit/Shared/XR/XRSystem.serialization.in:
* Source/WebKit/UIProcess/XR/PlatformXRSystem.cpp:
(WebKit::PlatformXRSystem::requestFrame):
* Source/WebKit/UIProcess/XR/PlatformXRSystem.h:
* Source/WebKit/UIProcess/XR/PlatformXRSystem.messages.in:
* Source/WebKit/UIProcess/XR/ios/PlatformXRARKit.mm:
(WebKit::ARKitCoordinator::renderLoop):
* Source/WebKit/UIProcess/XR/ios/WKARPresentationSession.h:
* Source/WebKit/UIProcess/XR/ios/WKARPresentationSession.mm:
(-[_WKARPresentationSession collectInputSources]):
(-[_WKTransientGestureRecognizer _platformXRInputSourceFromTransientAction:actionIdentifier:]):
(-[_WKTransientGestureRecognizer collectInputSources]):
* Source/WebKit/WebKit.xcodeproj/project.pbxproj:

Canonical link: https://commits.webkit.org/271912@main
